src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -218,14 +218,15 @@ static SourceGeneratorExecutionVersionMap FilterToProjectCone(SourceGeneratorExe
                 using var _5 = PooledHashSet<Checksum>.GetInstance(out var newChecksumsToSync);
                 newChecksumsToSync.AddRange(newProjectIdToChecksum.Values);
 
-                var newProjectStateChecksums = await _assetProvider.GetAssetsAsync<ProjectStateChecksums>(
-                    assetPath: AssetPath.SolutionAndTopLevelProjectsOnly, newChecksumsToSync, cancellationToken).ConfigureAwait(false);
```

Comment:
instead of having to allocate into this array, just to copy those all out and put into a dictionary below, we can now just read things right into the dictionary itself :)